trio/_highlevel_open_tcp_listeners.py

```diff
@@ -131,6 +146,16 @@ def _compute_backlog(backlog):
             listener.socket.close()
         raise
 
+    if unsupported_address_families and not listeners:
+        if len(unsupported_address_families) == 1:
+            raise unsupported_address_families[0]
+        else:
+            raise OSError(
+                errno.EAFNOSUPPORT,
+                "This system doesn't support any of the kinds of "
+                "socket that that address could use"
+            ) from trio.MultiError(unsupported_address_families)
```

Comment:
I guess this is fine but... it seems more complicated than it needs to be? Why not use the second branch unconditionally? (It would also simplify the tests.)

Reply:
Sure, I can make it always wrap the exception.